src/Modals/tests/AssignRoute.test.js

```diff
@@ -0,0 +1,73 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { mount } from 'enzyme';
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import AssignRouteModal, { assignRouteToBus } from '../AssignRoute.js';
+
+const mockUseLocationValue = {
+  pathname: '/testroute',
+  search: '',
+  hash: '',
+  state: {
+    plateNumber: 'RAB162H',
+    busType: 'coaster',
+    seats: 25,
+    route: {
+        origin: "Nyabugogo",
+        destination: "Remera"
+    }
+  }
+};
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: jest.fn().mockImplementation(() => {
+    return mockUseLocationValue;
+  })
+}));
+
+describe('Modal tests', () => {
+  const event = {
+    target: {
+      route: { value: 'some value' }
+    }
+  };
+  it('It should test the onSubmit function', () => {
+    const onSubmit = jest.fn();
+    const component = mount(
+      <MemoryRouter>
+        <AssignRouteModal onSubmit={onSubmit()} />
+      </MemoryRouter>
+    );
+    const form = component.find('form');
+    form.simulate('submit', event);
+    expect(onSubmit).toBeCalledTimes(1);
+  });
+  it('Should render the assign modal', async () => {
+    const onSubmitMock = jest.fn();
+    const handleCloseMock = jest.fn();
+    
+    const elem = render(
+      <MemoryRouter>
+        <AssignRouteModal
+          onSubmit={onSubmitMock}
+          handleClose={handleCloseMock}
+        ></AssignRouteModal>
+      </MemoryRouter>
+    );
+
+    fireEvent.change(screen.getByTestId('bus-route'), {
+        target: {
+            value: 'RN45'
+        }
+    });
+    await new Promise((r) => setTimeout(r, 500));
```

Comment:
These timeout are not needed since we are mocking the global axios it will be called **synchronously** which guarantee direct execution with no await during the tests